app/models/events/pull_request.rb

```diff
@@ -40,44 +40,96 @@ class PullRequest < ApplicationRecord
               :number,
               presence: true
     validates :draft,
-              :merged,
               :locked,
               inclusion: { in: [true, false] }
     validates :github_id, uniqueness: true
 
-    class << self
-      def resolve(payload)
-        action = payload[:action]
-        public_send(action, payload) if handleable?(action)
-      end
+    attr_accessor :payload
 
-      # Actions
+    def resolve
+      return unless handleable?
 
-      def merged(payload)
-        PullRequestJobs::Merged.perform_later(payload)
-      end
+      handle_action
+    end
 
-      def opened(payload)
-        PullRequestJobs::Opened.perform_later(payload)
-      end
+    def open
+      opened
+    end
 
-      def closed(payload)
-        PullRequestJobs::Closed.perform_later(payload)
-      end
+    def closed
+      close
```

Comment:
Why have this method instead of doing everything inside of the `closed` method? Same comment applies for the other actions.

Reply:
Just to keep a record of the methods on the public side, but it is ok to delete them.